### PR TITLE
Refactor lockers

### PR DESCRIPTION
**Pull request by zenangst**
This PR adds a `UserDefaultsLocker` which `KeychainLocker` inherits from.
Locker has been renamed to `KeychainLocker`.

- [x] Support migrations

### CODE REVIEW

File: Sources/Shared/KeychainLocker.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import Foundation
+import Keychain
+import Sugar
+
+public struct Keys {
```
Comment (vadymmarkov) [inline Sources/Shared/KeychainLocker.swift]: Could be internal I guess.

**Comment by zenangst** [comment]
One things that I want to add is migrations

File: Sources/Shared/KeychainLocker.swift (hunk 1 above)
Comment (zenangst) [inline Sources/Shared/KeychainLocker.swift]: Good point!

File: Sources/Shared/Lockable.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  public func migrate(from: Lockable, to: Lockable) {
+    to.accessToken = from.accessToken
+    to.refreshToken = from.refreshToken
+    to.tokenType = from.tokenType
+    to.expiryDate = from.expiryDate
+    to.userName = from.userName
+    to.userUPN = from.userUPN
+    from.clear()
+  }
```
Comment (zenangst) [inline Sources/Shared/Lockable.swift]: This should kinda work right :sunglasses:

File: Sources/Shared/KeychainLocker.swift (hunk 1 above)
Comment (zenangst) [inline Sources/Shared/KeychainLocker.swift]: Fixed :)

**Comment by zenangst** [comment]
The "tests" pass :grin:

File: Sources/Shared/Lockable.swift (hunk 2 above)
Comment (zenangst) [inline Sources/Shared/Lockable.swift]: Or this is a bit weird, it should be mutating an modify `self` right? :)

**Comment by vadymmarkov** [comment]
@zenangst :star2:

**Pull request merged**